src/osd/scheduler/mClockScheduler.cc

```diff
@@ -454,17 +468,63 @@ void mClockScheduler::enqueue(OpSchedulerItem&& item)
 
 void mClockScheduler::enqueue_front(OpSchedulerItem&& item)
 {
-  immediate.push_back(std::move(item));
+  unsigned priority = item.get_priority();
+  unsigned cutoff = get_io_prio_cut(cct);
+  auto id = get_scheduler_id(item);
+
```

Comment:
This is a bit odd -- ```priority >= cutoff```` but ```op_scheduler_class::immediate != id.class_id``` in enqueue would cause the op to be put into the high queue rather than the immediate queue.